server/src/main/java/org/apache/cassandra/sidecar/acl/authorization/FeaturePermission.java

```diff
@@ -0,0 +1,130 @@
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.cassandra.sidecar.acl.authorization;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import io.vertx.ext.auth.authorization.WildcardPermissionBasedAuthorization;
+import io.vertx.ext.auth.authorization.impl.WildcardPermissionBasedAuthorizationImpl;
+
+import static org.apache.cassandra.sidecar.acl.authorization.BasicPermissions.CREATE_RESTORE_JOB;
+import static org.apache.cassandra.sidecar.acl.authorization.BasicPermissions.CREATE_SNAPSHOT;
+import static org.apache.cassandra.sidecar.acl.authorization.BasicPermissions.DELETE_RESTORE_JOB;
+import static org.apache.cassandra.sidecar.acl.authorization.BasicPermissions.DELETE_SNAPSHOT;
+import static org.apache.cassandra.sidecar.acl.authorization.BasicPermissions.DELETE_STAGED_SSTABLE;
+import static org.apache.cassandra.sidecar.acl.authorization.BasicPermissions.EDIT_RESTORE_JOB;
+import static org.apache.cassandra.sidecar.acl.authorization.BasicPermissions.IMPORT_STAGED_SSTABLE;
+import static org.apache.cassandra.sidecar.acl.authorization.BasicPermissions.READ_GOSSIP;
+import static org.apache.cassandra.sidecar.acl.authorization.BasicPermissions.READ_RESTORE_JOB;
+import static org.apache.cassandra.sidecar.acl.authorization.BasicPermissions.READ_RING_KEYSPACE_SCOPED;
+import static org.apache.cassandra.sidecar.acl.authorization.BasicPermissions.READ_SCHEMA;
+import static org.apache.cassandra.sidecar.acl.authorization.BasicPermissions.READ_SCHEMA_KEYSPACE_SCOPED;
+import static org.apache.cassandra.sidecar.acl.authorization.BasicPermissions.READ_SNAPSHOT;
+import static org.apache.cassandra.sidecar.acl.authorization.BasicPermissions.READ_TOPOLOGY;
+import static org.apache.cassandra.sidecar.acl.authorization.BasicPermissions.STREAM_SNAPSHOT;
+import static org.apache.cassandra.sidecar.acl.authorization.BasicPermissions.UPLOAD_STAGED_SSTABLE;
+import static org.apache.cassandra.sidecar.acl.authorization.CassandraPermissions.MODIFY;
+import static org.apache.cassandra.sidecar.acl.authorization.CassandraPermissions.SELECT;
+import static org.apache.cassandra.sidecar.acl.authorization.ResourceScopes.TABLE;
+
+/**
+ * Enumerates a list of feature level permissions that Sidecar recognizes and honors.
+ */
+public enum FeaturePermission
+{
+    BULK_READ_DIRECT("ANALYTICS:READ_DIRECT",
```

Comment:
Can the enum match with the `name`? The enum is `BULK_READ_DIRECT`, but the name has `ANALYTICS`

Reply:
Yes, feature level permission name will be `ANALYTICS:READ_DIRECT` like other sidecar basic permissions, we can do following `ANALYTICS:READ_DIRECT,WRITE_DIRECT` or `ANALYTICS:*`. `BULK_READ_DIRECT` enum name, we currently don't use it. We can use it for name match, if we choose to go with `BULK_READ_DIRECT` for feature permission name.

Reply:
I was asking if it make sense to have this

```suggestion
    ANALYTICS_READ_DIRECT("ANALYTICS:READ_DIRECT",
```